x/bank/v2/keeper/keeper.go

```diff
@@ -38,3 +53,229 @@ func NewKeeper(authority []byte, addressCodec address.Codec, env appmodulev2.Env
 
 	return k
 }
+
+// MintCoins creates new coins from thin air and adds it to the module account.
+// An error is returned if the module account does not exist or is unauthorized.
+func (k Keeper) MintCoins(ctx context.Context, moduleName string, amounts sdk.Coins) error {
+
+	// TODO: Mint restriction
+	acc := k.ak.GetModuleAccount(ctx, moduleName)
```

Comment:
we are trying to remove auth from being a dependency. Im not sure we want to introduce it again here. The module account system is a legacy design we should not bring into0 bank v2 imo.

Reply:
then Mint will take input as `[]byte`. We should check if this address have `Minter` permission. Marked as TODO.
```go
if !acc.HasPermission(authtypes.Minter) {
```